repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java

```diff
@@ -998,15 +998,24 @@ protected void deleteVertex(AtlasVertex instanceVertex, boolean force) throws At
 
         // Delete external references to this vertex - incoming edges from lineage or glossary term edges
         final Iterable<AtlasEdge> incomingEdges    = instanceVertex.getEdges(AtlasEdgeDirection.IN);
+        final Iterable<AtlasEdge> outgoingEdges    = instanceVertex.getEdges(AtlasEdgeDirection.OUT);
         final boolean             isPurgeRequested = RequestContext.get().isPurgeRequested();
 
+        if (!RequestContext.get().getDeleteType().equals(DeleteType.SOFT)) {
```

Comment:
nit: We can simply add `isSoftDelete` or `isHardDelete` or `isPurge` etc with in `DeleteType` enum to void boiler plate code and re-write the code as `RequestContext.get().getDeleteType().isSoftDelete()`